### PR TITLE
Update deps, pip check, re-arch

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

- (maybe) addresses #50
- relaxes python pin to allow py38
- updates a number of dependency range changes from [upstream](https://github.com/tox-dev/tox/blob/3.14.3/setup.cfg#L40-L49)
- adds windows-only `colorama` dep, forcing removal of `noarch: python` :crying_cat_face: 
- tries `pip check` to catch future version drift
- adds myself to maintainer list

### CODE REVIEW

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

**Comment by CJ-Wright** [comment]
@conda-forge/tox is this ready?

**Comment by bollwyvl** [comment]
I feel like it is: I've been using a variant of this off-cf, and it's working fine...

**Comment by CJ-Wright** [comment]
@bollwyvl would you be interested in becoming a maintainer?

**Comment by bollwyvl** [comment]
Gah, I try to keep out of packages I am not a _frequent_ user of, but sure, given the thunderous response to this, I'll add myself...

File: recipe/meta.yaml [hunk 1]
```diff
@@ -53,9 +56,11 @@ about:
...
   doc_url: https://tox.readthedocs.org
   dev_url: https://github.com/tox-dev/tox
+  doc_source_url: https://github.com/tox-dev/tox/tree/3.14.3/docs
 
 extra:
   recipe-maintainers:
     - cshaley
     - kalefranz
     - sannykr
+    - bollwyvl
```
Comment (bollwyvl) [inline recipe/meta.yaml:66]: looks like i already did!

Comment (CJ-Wright) [reply recipe/meta.yaml:66]: :sheep: :man_facepalming:

**Comment by CJ-Wright** [comment]
Thank you very much!

**Comment by bollwyvl** [comment]
@CJ-Wright some azure builds failed on `master` with some early conda errors... trying out that guidance from @jakirkham on https://github.com/conda-forge/ipykernel-feedstock/pull/55 to get them to re-run... might have started too early...

**Comment by CJ-Wright** [comment]
Ok, let me know if you need me to do something

**Comment by bollwyvl** [comment]
Gah, same error:
```
An error occurred while installing package 'conda-forge::pkginfo-1.5.0.1-py_0'.
CondaError: Cannot link a source that does not exist. C:\Miniconda\Scripts\conda.exe
Running `conda clean --packages` may resolve your problem.
Attempting to roll back.
```

Doing some looking...

**Pull request merged**